sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -38,8 +38,20 @@ enum class graph_state {
   executable, ///< In executable state, the graph is ready to execute.
 };
 
-// Forward declaration
-class node;
+/// Class representing a node in the graph, returned by command_graph::add().
+class __SYCL_EXPORT node {
+private:
+  node(const std::shared_ptr<detail::node_impl> &Impl) : impl(Impl) {}
+
+  template <class Obj>
+  friend decltype(Obj::impl)
+  sycl::detail::getSyclObjImpl(const Obj &SyclObject);
+  template <class T>
+  friend T sycl::detail::createSyclObjFromImpl(decltype(T::impl) ImplObj);
+
+  std::shared_ptr<detail::node_impl> impl;
+  std::shared_ptr<detail::graph_impl> MGraph;
```

Comment:
Unrelated to this issue but since we're changing this code I think `MGraph` is unused so we could probably get rid of it.